config/routes.rb

```diff
@@ -13,6 +13,15 @@
   get '/users/:id', to: 'users#show' 
   # Get /users/id/posts
   get '/users/:user_id/posts', to: 'posts#index' 
+    # Get /users/id/posts/new
+  get '/users/:user_id/posts/new', to: 'posts#new' 
   # Get /users/user_id/posts/id
   get '/users/:user_id/posts/:id', to: 'posts#show' 
+  # like post
+  post '/users/:user_id/posts/:post_id/likes', to: 'likes#create', as: 'user_post_likes'
+  #add comment to post
+  post '/users/:user_id/posts/:post_id/comments', to: 'comments#create', as: 'user_post_comments'
+  resources :users do
+    resources :posts, only: [:index, :show, :new, :create]
+  end
 end
```

Comment:
- [ ] [OPTIONAL] I can see that you've started to use RESTful routes with the resources :users do ... end block, but you can extend this approach to include other resources like posts, likes, and comments. This can make your routes more conventional and easier to understand.

``` ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :likes, only: [:create]
      resources :comments, only: [:create, :new]
    end
end
```

Reply:
@Whoistolu 
Thank you for the feedbacks and insights
I have incorporated the required changes  🚀